x-pack/plugins/entity_manager/server/lib/v2/definitions/setup_entity_definitions_index.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient } from '@kbn/core/server';
+import { DEFINITIONS_ALIAS, TEMPLATE_VERSION } from '../constants';
+
+const definitionsIndexTemplate = {
+  name: `${DEFINITIONS_ALIAS}-template`,
+  index_patterns: [`${DEFINITIONS_ALIAS}-*`],
+  _meta: {
+    description: "Index template for the Elastic Entity Model's entity definitions index.",
+    managed: true,
+    managed_by: 'elastic_entity_model',
+  },
+  version: TEMPLATE_VERSION,
+  template: {
+    aliases: {
+      [DEFINITIONS_ALIAS]: {},
+    },
+    mappings: {
+      properties: {
+        template_version: {
+          type: 'short',
+        },
+        definition_type: {
+          type: 'keyword',
+        },
+      },
+    },
+  },
+};
+
+const CURRENT_INDEX = `${DEFINITIONS_ALIAS}-${TEMPLATE_VERSION}` as const;
+
+export async function setupEntityDefinitionsIndex(clusterClient: IClusterClient) {
+  const esClient = clusterClient.asInternalUser;
+
+  await esClient.indices.putIndexTemplate(definitionsIndexTemplate);
+
+  const indices = await esClient.indices.get({
+    index: `${DEFINITIONS_ALIAS}-*`,
+  });
+  const indexNames = Object.keys(indices);
+
+  if (indexNames.includes(CURRENT_INDEX)) {
+    return;
```

Comment:
can we directly target the current_index instead of the wildcard lookup ? can we also log whether the index is pre existing or not

Reply:
> can we directly target the current_index instead of the wildcard lookup ?

What do you aim to gain from that? The benefit of the wildcard lookup is that ES doesn't throw a 404 if the current index isn't installed, but if some older version is installed we still get that back and can act on that (though we currently don't do that, I had a version where I did but decided to cut that out).

> can we also log whether the index is pre existing or not, and the template version installed

Yes!

Reply:
> What do you aim to gain from that

just easier to figure out what happens, I was wondering why we needed to resolve the wildcard